### PR TITLE
Add ControllerNodeId and SelectionReason to StartSession.

### PR DESCRIPTION
**Pull request by boks1971**
Media node will have that information with this change and can log it in context.

Also switching WS client open/close to debug level log.

### CODE REVIEW

File: pkg/routing/redisrouter.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ func (r *RedisRouter) StartParticipantSignal(ctx context.Context, roomName livek
 		return
 	}
 
-	return r.StartParticipantSignalWithNodeID(ctx, roomName, pi, livekit.NodeID(rtcNode.Id))
+	return r.StartParticipantSignalWithNodeID(ctx, roomName, pi, livekit.NodeID(rtcNode.Id), "selector")
```
Comment (boks1971) [inline pkg/routing/redisrouter.go:164]: Just marking it as "selector" as the `selectionReason` for now. In the future, can change `Selector` interface `SelectNode` method to return a proper reason and use that.

**Review by paulwe — APPROVED** [review]
is this relevant to oss? should we pass this to cloud using psrpc metadata like we do with the project id?

**Comment by boks1971** [comment]
> is this relevant to oss? should we pass this to cloud using psrpc metadata like we do with the project id?

Not in local router case definitely. But, it could be for redis + custom selector case. But, it feels like passing in via PsRPC would be good and not disturb things here (other than log level change). Can you point me to how it is done for projectID? I will have a look and revert these changes (including protocol change) if that works out better.

**Comment by boks1971** [comment]
> > is this relevant to oss? should we pass this to cloud using psrpc metadata like we do with the project id?
> 
> Not in local router case definitely. But, it could be for redis + custom selector case. But, it feels like passing in via PsRPC would be good and not disturb things here (other than log level change). Can you point me to how it is done for projectID? I will have a look and revert these changes (including protocol change) if that works out better.

Found how it is done. Will take a look at that.

**Review by cnderrauber — APPROVED** [review]

File: pkg/service/signal_test.go [hunk 2]
```diff
@@ -70,6 +70,8 @@ func TestSignal(t *testing.T) {
 				roomName livekit.RoomName,
 				pi routing.ParticipantInit,
 				connectionID livekit.ConnectionID,
+				controllerNodeID livekit.NodeID,
+				selectionReason string,
```
Comment (paulwe) [inline pkg/service/signal_test.go:74]: still needed?

Comment (boks1971) [reply pkg/service/signal_test.go:74]: Nah, cleaning up :-)

**Pull request merged**